salt/_modules/etcd_discovery_helpers.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import absolute_import
+import logging
+
+
+LOG = logging.getLogger(__name__)
+
+
+def __virtual__():
+    return "etcd_discovery_helpers"
+
+
+def get_etcd_cluster_size():
+    """
+    Determines the etcd discovery cluster size
+
+    Determines the desired number of cluster members, defaulting to
+    the value supplied in the etcd:masters pillar, falling back to
+    match the number nodes with the kube-master role, and if this is
+    less than 3, it will bump it to 3.
+    """
+    member_count = __pillar__["etcd"]["masters"]
+
+    if member_count is not None:
+        member_count = int(member_count)
+    else:
+        member_count = len(__salt__['mine.get'](
+            'roles:kube-master', 'fqdn', expr_form='grain').values())
+
+    if member_count < 3:
+        LOG.warning("etcd member count too low (%d), increasting to 3",
+                    member_count)
+        member_count = 3
```

Comment:
Does it make sense to force `member_count = 3` when we have set `pillar["etcd"]["masters"]=1`? Maybe we know what we are doing...